### ISSUE TITLE
[Feature request] ~ in --sub-paths

### ISSUE THREAD

**Issue by rr-** [mpv-player/mpv#4469]
### mpv version and platform

```
mpv 0.25.0-92-gbc3365b16 (C) 2000-2017 mpv/MPlayer/mplayer2 projects
 built on Tue May 30 00:42:02 CEST 2017
ffmpeg library versions:
   libavutil       55.58.100
   libavcodec      57.89.100
   libavformat     57.71.100
   libswscale      4.6.100
   libavfilter     6.82.100
   libswresample   2.7.100
ffmpeg version: 3.3.1
```

### Reproduction steps

Try to use `sub-paths=~/subs` in the config file.

### Expected behavior

mpv expands `~/subs` to `$HOME/subs` == `/home/rr-/subs` and discovers the files in that directory.

### Actual behavior

mpv doesn't load the subtitles until I explicitly set `/home/rr-/subs` as `sub-paths` option. I think what happens is that mpv looks in `./\~/subs` instead.

### Log file

I don't think it is needed.

### Sample files

I don't think it is needed.

**Comment by wm4** [comment]
Do you want to implement it? Should be simple enough.

**Comment by rr-** [comment]
Sure. Can I do it in the place that reads the option's value, or should I create a generic mechanism for all path-based options?

**Comment by wm4** [comment]
> Sure. Can I do it in the place that reads the option's value, 

Yes.

> or should I create a generic mechanism for all path-based options?

No, doing that during option parsing would result in an unholy mess.

**Comment by wiiaboo** [comment]
Merged in 1a25f8c.
